kbase-extension/static/kbase/js/common/jobManager.js

```diff
@@ -323,6 +321,131 @@ define(['common/jobMessages', 'common/jobs'], (JobMessages, Jobs) => {
         }
     }
 
+    /**
+     * A set of generic message handlers
+     *
+     * @param {class} Base
+     * @returns
+     */
+    const defaultHandlerMixin = (Base) =>
+        class extends Base {
+            /**
+             * parse job info and update the appropriate part of the model
+             *
+             * @param {object} message
+             */
+            handleJobInfo(self, message) {
+                const { jobInfo, error } = message;
+                if (error) {
+                    return;
+                }
+
+                self.model.setItem(`exec.jobs.params.${jobInfo.job_id}`, jobInfo.job_params[0]);
+                self.removeListener(jobInfo.job_id, 'job-info');
+            }
+
+            handleJobRetry(self, message) {
+                const { job, retry, error } = message;
+                if (error) {
+                    return;
+                }
+
+                // copy over the params
+                self.model.setItem(
+                    `exec.jobs.params.${retry.jobState.job_id}`,
+                    self.model.getItem(`exec.jobs.params.${job.jobState.job_id}`)
+                );
+
+                // request job updates for the new job
+                self.addListener('job-status', [retry.jobState.job_id]);
+                self.bus.emit('request-job-updates-start', {
+                    jobId: retry.jobState.job_id,
+                });
+                // update the model with the job data
+                self.updateModel(
+                    [job, retry].map((j) => {
+                        return j.jobState;
+                    })
+                );
+            }
+
+            handleJobDoesNotExist(self, message) {
+                const { jobId } = message;
+                self.handleJobStatus(self, {
+                    jobId,
+                    jobState: {
+                        job_id: jobId,
+                        status: 'does_not_exist',
+                    },
+                });
```

Comment:
I am a little concerned about this as it seems as though there may be occasions where the FE gets an incorrect 'job-does-not-exist' message from the backend. Will investigate further once the job status table is updated.

Reply:
That sounds problematic. We should know what causes that.